File: problem_solving_for_beginners/python/prime_number.py
# checks for a prime number 
def isprime(num):
    flag = 0
    for i in range(2, num):
        if num % i == 0:
            flag = 1
            break
    return flag

def print_prime(flag):
    if flag == 0:
        print("Prime number")
    else:
        print("Not a Prime number")
        

num = int(input("Enter a positive number: "))
flag = isprime(num)
print_prime(flag)

File: problem_solving_for_beginners/c_programming/prime_number.c
#include<stdio.h>
/**
 * isprime - checks for a prime number
 * 
 * @num: number to check
 * 
 * Return: 0 (prime), 1 (Not Prime)
 */
int isprime(int num)
{
    int i, flag;

    for (i = 2; i < num; i++)
    {
        if (num % i == 0)
        {
            flag = 1;
            break;
        }
        else
            flag = 0;
    }
    return (flag);
}

/**
 * print_prime - prints prime number if flag is 0 Else  Not prime number
 * @flag: number flag
 */
void print_prime(int flag)
{
    if (flag == 0)
        printf("Prime number\n");
    else
        printf("Not a Prime number\n");
}
/**
 * main - finds a factor of a number
 *
 * Return: 0 always (success)
 */
int main(void)
{
    int num, flag;

    printf("Enter a number: ");
    scanf("%d", &num);

    flag = isprime(num);
    print_prime(flag);

    return 0;
}